sql/src/test/java/io/crate/testing/LuceneDocCollectorProvider.java

```diff
@@ -0,0 +1,129 @@
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.testing;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterables;
+import io.crate.action.job.SharedShardContexts;
+import io.crate.analyze.*;
+import io.crate.analyze.relations.PlannedAnalyzedRelation;
+import io.crate.breaker.RamAccountingContext;
+import io.crate.jobs.JobContextService;
+import io.crate.jobs.JobExecutionContext;
+import io.crate.metadata.Routing;
+import io.crate.operation.collect.CrateCollector;
+import io.crate.operation.collect.JobCollectContext;
+import io.crate.operation.collect.MapSideDataCollectOperation;
+import io.crate.operation.projectors.RowReceiver;
+import io.crate.planner.Planner;
+import io.crate.planner.consumer.ConsumerContext;
+import io.crate.planner.consumer.QueryAndFetchConsumer;
+import io.crate.planner.node.dql.CollectAndMerge;
+import io.crate.planner.node.dql.CollectPhase;
+import io.crate.sql.parser.SqlParser;
+import org.elasticsearch.common.breaker.CircuitBreaker;
+import org.elasticsearch.common.breaker.NoopCircuitBreaker;
+import org.elasticsearch.index.IndexService;
+import org.elasticsearch.indices.IndexMissingException;
+import org.elasticsearch.indices.IndicesService;
+import org.elasticsearch.test.InternalTestCluster;
+
+import java.util.*;
+
+/**
+ * Create preconfigured CrateCollectors from a SQL statement and a TestCluster.
+ * Returns all collectors from all TestCluster nodes.
+ */
+public class LuceneDocCollectorProvider implements AutoCloseable {
+
+    private static final RamAccountingContext RAM_ACCOUNTING_CONTEXT =
+            new RamAccountingContext("dummy", new NoopCircuitBreaker(CircuitBreaker.Name.FIELDDATA));
+
+    private final InternalTestCluster cluster;
+    private final Analyzer analyzer;
+    private final QueryAndFetchConsumer queryAndFetchConsumer;
+
+    private List<JobCollectContext> collectContexts = new ArrayList<>();
+
+    public LuceneDocCollectorProvider(InternalTestCluster cluster) {
+        this.cluster = cluster;
+        this.analyzer = cluster.getDataNodeInstance(Analyzer.class);
+        this.queryAndFetchConsumer = cluster.getDataNodeInstance(QueryAndFetchConsumer.class);
+    }
+
+    private Iterable<CrateCollector> createNodeCollectors(String nodeId, String indexName, CollectPhase collectPhase, RowReceiver downstream) {
+        IndexService indexService;
+        IndicesService indicesService;
+        String nodeName = cluster.clusterService().state().nodes().get(nodeId).name();
+        try {
+            indicesService = cluster.getInstance(IndicesService.class, nodeName);
+            indexService = indicesService.indexServiceSafe(indexName);
+        } catch (IndexMissingException e) {
+            return Collections.emptyList();
+        }
+
+        JobContextService jobContextService = indexService.shardInjectorSafe(0).getInstance(JobContextService.class);
```

Comment:
there is no need to retrieve the JobContextService from a shardInjector.. can be retrieved from the dataNode globally